File: streamscale_fe/web/src/types/video.ts
export type VideoType = 'MOVIE' | 'SERIES';
export type VideoStatus = 'DRAFT' | 'UPLOADED' | 'READY' | 'PUBLISHED';

export interface Genre {
  id: number;
  name: string;
}

export interface Video {
  id: number;
  title: string;
  description: string | null;
  type: VideoType;
  status: VideoStatus;
  thumbnailUrl: string | null;
  trailerUrl: string | null;
  durationSeconds: number | null;
  releaseYear: number | null;
  originalFileUrl?: string | null;
  originalFileName?: string | null;
  originalFileSize?: number | null;
  genres: Genre[];
}

export interface VideoUpdateRequest {
  title?: string;
  description?: string;
  type?: VideoType;
  status?: VideoStatus;
  thumbnailUrl?: string;
  trailerUrl?: string;
  durationSeconds?: number | null;
  releaseYear?: number | null;
  genreIds?: number[];
}

export interface VideoCreateRequest {
  title: string;
  description?: string;
  type: VideoType;
  status: VideoStatus;
  thumbnailUrl?: string;
  trailerUrl?: string;
  durationSeconds?: number;
  releaseYear?: number;
  genreIds?: number[];
}

File: streamscale_be/src/main/java/com/micheal/streamscale/catalog/entity/Video.java
package com.micheal.streamscale.catalog.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "videos")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Video {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private VideoType type;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private VideoStatus status;

    @Column(name = "thumbnail_url", columnDefinition = "TEXT")
    private String thumbnailUrl;

    @Column(name = "trailer_url", columnDefinition = "TEXT")
    private String trailerUrl;

    @Column(name = "duration_seconds")
    private Integer durationSeconds;

    @Column(name = "release_year")
    private Integer releaseYear;

    @ManyToMany
    @JoinTable(
            name = "video_genres",
            joinColumns = @JoinColumn(name = "video_id"),
            inverseJoinColumns = @JoinColumn(name = "genre_id")
    )
    @Builder.Default
    private Set<Genre> genres = new HashSet<>();

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();

        if (status == null) {
            status = VideoStatus.DRAFT;
        }
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}